scripts/bqeth_subscription.py

```diff
@@ -0,0 +1,267 @@
+@setup_context
+def pay_subscription(context, encryptor_slots):
+    """Pay for a new subscription period and initial encryptor slots."""
+    base_fees = context.subscription_contract.functions.baseFees(0).call()
+    encryptor_fees = context.subscription_contract.functions.encryptorFees(
+        MAX_NODES, context.subscription_contract.functions.subscriptionPeriodDuration().call()
+    ).call()
+
+    click.echo(
+        f"Approving transfer of {base_fees + encryptor_fees} ERC20 token for subscription contract."
+    )
+    tx_hash = context.erc20_contract.functions.approve(
+        SUBSCRIPTION_CONTRACT_ADDRESS, base_fees + encryptor_fees
+    ).transact({"from": context.account.address})
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+    click.echo(f"Paying for a new subscription period with {encryptor_slots} encryptor slots.")
+    tx_hash = context.subscription_contract.functions.payForSubscription(encryptor_slots).transact(
+        {"from": context.account.address}
+    )
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+
+@cli.command()
+@click.option("--extra-slots", default=1, help="Number of additional encryptor slots to pay for.")
+@setup_context
+def pay_slots(context, extra_slots):
+    """Pay for additional encryptor slots."""
+    click.echo(f"Paying for {extra_slots} new encryptor slots.")
+    tx_hash = context.subscription_contract.functions.payForEncryptorSlots(extra_slots).transact(
+        {"from": context.account.address}
+    )
+    click.echo(f"Transaction hash: {tx_hash.hex()}")
+
+
+@cli.command()
+@click.option("--num-nodes", default=2, help="Number of nodes to use for the ritual.")
+@click.option("--duration", help="Duration in seconds for the ritual.", type=int)
+@click.option("--random-seed", help="Random seed integer for sampling.")
+@click.option("--num-rituals", default=1, help="Number of rituals to initiate.", type=int)
+@setup_context
+def initiate_ritual(context, num_nodes, duration, random_seed, num_rituals):
+    """Initiate a ritual."""
+    if NUCYPHER_DOMAIN == "mainnet":
+        porter_request = f"https://porter.nucypher.community/bucket_sampling?quantity={num_nodes}&random_seed={random_seed}"
+    else:
+        porter_request = f"{PORTER_ENDPOINT}/get_ursulas?quantity={num_nodes}"
+
+    # start_of_subscription = context.subscription_contract.functions.startOfSubscription().call()
+    # duration = (
+    #     context.subscription_contract.functions.subscriptionPeriodDuration().call()
+    #     + context.subscription_contract.functions.yellowPeriodDuration().call()
+    #     + context.subscription_contract.functions.redPeriodDuration().call()
+    # )
+    # if start_of_subscription > 0:
+    #     click.echo(
+    #         "Subscription has already started. Subtracting the elapsed time from the duration."
+    #     )
+    #     now = context.w3.eth.get_block("latest")["timestamp"]
+    #     elapsed = now - start_of_subscription + 100
+    #     duration -= elapsed
```

Comment:
Why is this block commented out?

Reply:
i think some commit have been pushed that shouldn't after the script was used locally for some testing